src/Microsoft.CSharp/tests/DynamicDebuggerProxyTests.cs

```diff
@@ -0,0 +1,137 @@
+        }
+
+        [Fact]
+        public void Items()
+        {
+            var obj = new DynamicTestObject(new Dictionary<string, object>() { { "F", 1 }, { "G", "" } });
+            var debugView = CreateDebugView(obj);
+            var items = GetItems(debugView);
+            Assert.Equal(2, items.Length);
+        }
+
+        [Fact]
+        public void TryGetMemberValue()
+        {
+            var obj = new DynamicTestObject(new Dictionary<string, object>() { { "F", 1 } });
+            Assert.Equal(1, TryGetMemberValue(obj, "F", ignoreException: false));
+        }
+
+        [Fact]
+        public void TryGetMemberValue_BindingFailed()
+        {
+            var obj = new DynamicTestObject(new Dictionary<string, object>());
+
+            // ignoreException: true
+            Assert.Null(TryGetMemberValue(obj, "F", ignoreException: true));
+
+            // ignoreException: false
+            var exceptionType = GetType("DynamicBindingFailedException");
+            Assert.Throws(exceptionType, () => GetItems(TryGetMemberValue(obj, "F", ignoreException: false)));
+        }
+
+        [Fact]
+        public void TryGetMemberValue_MissingMember()
+        {
+            var obj = new DynamicTestObject(new Dictionary<string, object>(), throwIfMissing: true);
+
+            // ignoreException: true
+            string message = (string)TryGetMemberValue(obj, "F", ignoreException: true);
+            Assert.NotNull(message);
+
+            // ignoreException: false
+            Assert.Throws<MissingMemberException>(() => GetItems(TryGetMemberValue(obj, "F", ignoreException: false)));
+        }
+
+        private static Type GetType(string typeName)
+        {
+            return typeof(Binder).Assembly.GetType("Microsoft.CSharp.RuntimeBinder." + typeName, throwOnError: true);
+        }
+
+        private static object CreateDebugView(object arg)
+        {
+            return Activator.CreateInstance(_debugViewType, new object[] { arg });
+        }
+
+        private static object[] GetItems(object debugView)
+        {
+            var method = _debugViewType.GetMethod("get_Items", BindingFlags.NonPublic | BindingFlags.Instance);
+            return (object[])InvokeAndUnwrapException(method, debugView, new object[0]);
+        }
+
+        private static object TryGetMemberValue(object obj, string name, bool ignoreException)
+        {
+            var method = _debugViewType.GetMethod("TryGetMemberValue", BindingFlags.NonPublic | BindingFlags.Static);
+            return InvokeAndUnwrapException(method, null, new object[] { obj, name, ignoreException });
+        }
+
+        private static object InvokeAndUnwrapException(MethodInfo method, object obj, object[] args)
+        {
+            try
+            {
+                return method.Invoke(obj, args);
+            }
+            catch (TargetInvocationException e)
+            {
+                throw e.InnerException;
+            }
```

Comment:
This is ok, but you should also be able to use BindingFlags.DoNotWrapExceptions.

Reply:
It looks like `BindingFlags.DoNotWrapExceptions` is not available on all platforms.

---
In reply to: [295053779](https://github.com/dotnet/corefx/pull/38656#discussion_r295053779) [](ancestors = 295053779)